### PR TITLE
Stop using unwrap_or_default in e2e tests

### PR DESCRIPTION
**Pull request by andresilva91**
## Motivation

`unwrap_or_default` can hide errors

## Proposal

Remove it

## Test Plan

CI

### CODE REVIEW

**Comment by andresilva91** [comment]
* **#1847** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1847?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @andresilva91 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: linera-service/tests/end_to_end_tests.rs [hunk 1]
```diff
@@ -45,8 +45,21 @@ impl FungibleApp {
...
         let response_body = self.0.query(&query).await.unwrap();
-        serde_json::from_value(response_body["accounts"]["entry"]["value"].clone())
-            .unwrap_or_default()
+        let deserialized_amount =
+            serde_json::from_value::<Amount>(response_body["accounts"]["entry"]["value"].clone());
+        match deserialized_amount {
+            Ok(amount) => amount,
+            Err(e)
+                if e.to_string()
+                    .contains("invalid type: null, expected a string")
```
Comment (ma2bd) [inline linera-service/tests/end_to_end_tests.rs:54]: This one is a bit too fragile for me. How about `let amount_option = serde_json::from_value::<Option<Amount>>(..).unwrap();` ?

Comment (andresilva91) [reply linera-service/tests/end_to_end_tests.rs]: Ah yeah, much better

File: linera-service/tests/end_to_end_tests.rs [hunk 2]
```diff
@@ -59,7 +63,7 @@ impl FungibleApp {
     async fn entries(&self) -> Vec<native_fungible::AccountEntry> {
         let query = "accounts { entries { key, value } }";
         let response_body = self.0.query(&query).await.unwrap();
-        serde_json::from_value(response_body["accounts"]["entries"].clone()).unwrap_or_default()
+        serde_json::from_value(response_body["accounts"]["entries"].clone()).unwrap()
```
Comment (ma2bd) [inline linera-service/tests/end_to_end_tests.rs:66]: You mean to give a type hint: `serde_json::from_value::<Option<_>>(response_body["accounts"]["entries"].clone()).unwrap()`

Comment (andresilva91) [reply linera-service/tests/end_to_end_tests.rs:66]: IIRC this should return an empty `Vec` when there are no entries, so we should be good here. If it throws, it's an actual error

**Review by ma2bd — APPROVED** [review]

**Review by afck — APPROVED** [review]

**Pull request merged**